src/components/AddressSearch.js

```diff
@@ -39,55 +39,53 @@ const defaultProps = {
+        if (!addressComponents) {
+            return;
+        }
+
+        // Gather the values from the Google details
+        const streetNumber = GooglePlacesUtils.getAddressComponent(addressComponents, 'street_number', 'long_name') || '';
+        const streetName = GooglePlacesUtils.getAddressComponent(addressComponents, 'route', 'long_name') || '';
+        const addressStreet = `${streetNumber} ${streetName}`.trim();
+        let addressCity = GooglePlacesUtils.getAddressComponent(addressComponents, 'locality', 'long_name');
+        if (!addressCity) {
+            addressCity = GooglePlacesUtils.getAddressComponent(addressComponents, 'sublocality', 'long_name');
+            Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: addressCity});
+        }
+        const addressZipCode = GooglePlacesUtils.getAddressComponent(addressComponents, 'postal_code', 'long_name');
+        const addressState = GooglePlacesUtils.getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
 
-            // Trigger text change events for each of the individual fields being saved on the server
-            props.onChangeText('addressStreet', `${streetNumber} ${streetName}`);
-            props.onChangeText('addressCity', city);
-            props.onChangeText('addressState', state);
-            props.onChangeText('addressZipCode', zipCode);
-        } else {
-            // Clear the values associated to the address, so our validations catch the problem
-            Log.hmmm('[AddressSearch] Search result failed validation: ', {
-                address: details.formatted_address,
-                address_components: addressComponents,
-                place_id: details.place_id,
-            });
-            props.onChangeText('addressStreet', null);
-            props.onChangeText('addressCity', null);
-            props.onChangeText('addressState', null);
-            props.onChangeText('addressZipCode', null);
+        const values = {};
+        if (addressStreet && addressStreet.length > props.value.length) {
+            // Don't replace if the user has typed something longer. I.e. maybe the user entered the Apt #
+            values.addressStreet = addressStreet;
         }
+        if (addressCity) {
+            values.addressCity = addressCity;
+        }
+        if (addressZipCode) {
+            values.addressZipCode = addressZipCode;
+        }
+        if (addressState) {
+            values.addressState = addressState;
+        }
+        if (_.size(values) === 0) {
+            return;
+        }
+        props.onChange(values);
     };
 
     return (
         <GooglePlacesAutocomplete
-            ref={googlePlacesRef}
             fetchDetails
             suppressDefaultStyles
             enablePoweredByContainer={false}
+            placeholder=""
```

Comment:
Why do we need an empty placeholder?

Reply:
@nickmurray47 's IDE was complaining about it: https://github.com/Expensify/App/pull/6200#discussion_r753477585

Looking at the [library's code](https://github.com/FaridSafi/react-native-google-places-autocomplete/blob/47d7223dd48f85da97e80a0729a985bbbcee353f/GooglePlacesAutocomplete.d.ts#L409), it is NOT an optional prop, so I figured we should add it.

Reply:
:O you are right, there is a default there. My IDE didn't complain either, maybe @nickmurray47 has something different in his environment. Considering there is a default, I'll remove it.